vault/vault.go

```diff
@@ -0,0 +1,335 @@
+package vault
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/AlexsJones/gravitywell/configuration"
+	"github.com/AlexsJones/gravitywell/vcs"
+	"github.com/fatih/color"
+	v1 "k8s.io/api/core/v1"
+	rbacv1beta1 "k8s.io/api/rbac/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func UnSetVaultConfiguration(opt configuration.Options, cluster configuration.ProviderCluster) error {
```

Comment:
Please, can you refactor this function? it is really big, it is really difficult to read it